src/node_v8.cc

```diff
@@ -45,6 +62,30 @@ void UpdateHeapStatisticsArrayBuffer(const FunctionCallbackInfo<Value>& args) {
 }
 
 
+void UpdateHeapSpaceStatisticsBuffer(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  if (args.Length() < 1)
+    return env->ThrowTypeError("heap space index is required");
+  if (!args[0]->IsNumber())
+    return env->ThrowTypeError("heap space index must be a number");
+
+  HeapSpaceStatistics s;
+  Isolate* isolate = env->isolate();
+  isolate->GetHeapSpaceStatistics(&s, args[0]->NumberValue());
+  uint32_t* const buffer = env->heap_space_statistics_buffer();
+#define V(index, name, _) buffer[index] = static_cast<uint32_t>(s.name());
+  HEAP_SPACE_STATISTICS_PROPERTIES(V)
+#undef V
+
+  Local<String> heap_space_name = String::NewFromUtf8(isolate,
+                                                      s.space_name(),
+                                                      NewStringType::kNormal)
+                                      .ToLocalChecked();
```

Comment:
I'd export the heap space names upfront; creating a new string every time is not very efficient.